Build failure in katip with mtl-2.3

```
Building library for katip-0.8.7.1..
[1 of 5] Compiling Katip.Core       ( src/Katip/Core.hs, dist/build/Katip/Core.o, dist/build/Katip/Core.dyn_o )
[2 of 5] Compiling Katip.Format.Time ( src/Katip/Format/Time.hs, dist/build/Katip/Format/Time.o, dist/build/Katip/Format/Time.dyn_o )
[3 of 5] Compiling Katip.Monadic    ( src/Katip/Monadic.hs, dist/build/Katip/Monadic.o, dist/build/Katip/Monadic.dyn_o )

src/Katip/Monadic.hs:372:7: error:
    Not in scope: type constructor or class ‘MonadPlus’
    |
372 |       MonadPlus,
    |       ^^^^^^^^^

src/Katip/Monadic.hs:375:7: error:
    Not in scope: type constructor or class ‘MonadFix’
    Perhaps you meant ‘MonadFail’ (imported from Prelude)
    |
375 |       MonadFix,
    |       ^^^^^^^^

src/Katip/Monadic.hs:486:7: error:
    Not in scope: type constructor or class ‘MonadPlus’
    |
486 |       MonadPlus,
    |       ^^^^^^^^^

src/Katip/Monadic.hs:488:7: error:
    Not in scope: type constructor or class ‘MonadFix’
    Perhaps you meant ‘MonadFail’ (imported from Prelude)
    |
488 |       MonadFix,
    |       ^^^^^^^^
```

As a Hackage trustee, I have revised the affected versions on Hackage. See e.g. https://hackage.haskell.org/package/katip-0.8.7.1/revisions/.

--- a/katip/src/Katip/Monadic.hs
+++ b/katip/src/Katip/Monadic.hs
@@ -43,11 +43,13 @@ where
 -------------------------------------------------------------------------------
 import Control.Applicative
 import Control.Exception.Safe
+import Control.Monad                               as CM
 import Control.Monad.Base
 import Control.Monad.Error.Class
 #if MIN_VERSION_base(4, 9, 0)
 import qualified Control.Monad.Fail                as MF
 #endif
+import Control.Monad.Fix                           as MFix
 import Control.Monad.IO.Class
 import Control.Monad.IO.Unlift
 import Control.Monad.Reader
@@ -71,7 +73,8 @@ import qualified Control.Monad.Trans.Writer.Strict as Strict
   ( WriterT,
     mapWriterT,
   )
-import Control.Monad.Writer hiding ((<>))
+import qualified Control.Monad.Writer as Lazy (WriterT, mapWriterT)
+import Control.Monad.Writer.Class as WC
 import Data.Aeson
 #if MIN_VERSION_aeson(2, 0, 0)
 import qualified Data.Aeson.KeyMap as KM
@@ -233,11 +236,11 @@ instance (Monoid w, KatipContext m, Katip (Strict.WriterT w m)) => KatipContext 
   getKatipNamespace = lift getKatipNamespace
   localKatipNamespace = Strict.mapWriterT . localKatipNamespace
 
-instance (Monoid w, KatipContext m, Katip (WriterT w m)) => KatipContext (WriterT w m) where
+instance (Monoid w, KatipContext m, Katip (Lazy.WriterT w m)) => KatipContext (Lazy.WriterT w m) where
   getKatipContext = lift getKatipContext
-  localKatipContext = mapWriterT . localKatipContext
+  localKatipContext = Lazy.mapWriterT . localKatipContext
   getKatipNamespace = lift getKatipNamespace
-  localKatipNamespace = mapWriterT . localKatipNamespace
+  localKatipNamespace = Lazy.mapWriterT . localKatipNamespace
 
 instance (Monoid w, KatipContext m, Katip (Strict.RWST r w s m)) => KatipContext (Strict.RWST r w s m) where
   getKatipContext = lift getKatipContext
@@ -367,12 +370,12 @@ newtype KatipContextT m a = KatipContextT
       MonadMask,
       MonadBase b,
       MonadState s,
-      MonadWriter w,
+      WC.MonadWriter w,
       MonadError e,
-      MonadPlus,
+      CM.MonadPlus,
       MonadResource,
       Alternative,
-      MonadFix,
+      MFix.MonadFix,
       MonadTrans
     )
 
@@ -481,11 +484,11 @@ newtype NoLoggingT m a = NoLoggingT
       MonadMask,
       MonadBase b,
       MonadState s,
-      MonadWriter w,
+      WC.MonadWriter w,
       MonadError e,
-      MonadPlus,
+      CM.MonadPlus,
       Alternative,
-      MonadFix,
+      MFix.MonadFix,
       MonadReader r
     )
 

